ding/example/ppo_with_complex_obs.py

```diff
@@ -0,0 +1,197 @@
+my_env_ppo_create_config = dict(
+    env_manager=dict(type='base'),
+    policy=dict(type='ppo'),
+)
+my_env_ppo_create_config = EasyDict(my_env_ppo_create_config)
+create_config = my_env_ppo_create_config
+
+
+class MyEnv(gym.Env):
+
+    def __init__(self, seq_len=5, feature_dim=10, image_size=(10, 10, 3)):
+        super().__init__()
+
+        # Define the action space
+        self.action_space = spaces.Discrete(2)
+
+        # Define the observation space
+        self.observation_space = spaces.Dict(
+            (
+                {
+                    '0': spaces.Dict(
+                        {
+                            'k1': spaces.Box(low=0, high=np.inf, shape=(1, ), dtype=np.float32),
+                            'k2': spaces.Box(low=-1, high=1, shape=(1, ), dtype=np.float32),
+                        }
+                    ),
+                    '1': spaces.Box(low=-np.inf, high=np.inf, shape=(seq_len, feature_dim), dtype=np.float32),
+                    '2': spaces.Box(low=0, high=255, shape=image_size, dtype=np.uint8),
+                    '3': spaces.Box(low=0, high=np.array([np.inf, 3]), shape=(2, ), dtype=np.float32)
+                }
+            )
+        )
+
+    def reset(self):
+        # Generate a random initial state
+        return self.observation_space.sample()
+
+    def step(self, action):
+        # Compute the reward and done flag (which are not used in this example)
+        reward = np.random.uniform(low=0.0, high=1.0)
+
+        done = False
+        if np.random.uniform(low=0.0, high=1.0) > 0.7:
+            done = True
+
+        info = {}
+
+        # Return the next state, reward, and done flag
+        return self.observation_space.sample(), reward, done, info
+
+
+def ding_env_maker():
+    return DingEnvWrapper(
+        MyEnv(), cfg={'env_wrapper': [
+            lambda env: EvalEpisodeReturnEnv(env),
+        ]}
+    )
+
+
+class Encoder(nn.Module):
+
+    def __init__(self, feature_dim: int):
+        super(Encoder, self).__init__()
+
+        # Define the networks for each input type
+        self.fc_net_1_k1 = nn.Sequential(nn.Linear(1, 8), nn.ReLU())
+        self.fc_net_1_k2 = nn.Sequential(nn.Linear(1, 8), nn.ReLU())
+        self.fc_net_1 = nn.Sequential(nn.Linear(16, 32), nn.ReLU())
+
+        self.class_token = nn.Parameter(torch.zeros(1, 1, feature_dim))
+        self.encoder_layer = nn.TransformerEncoderLayer(d_model=feature_dim, nhead=2, batch_first=True)
+        self.transformer_encoder = nn.TransformerEncoder(self.encoder_layer, num_layers=1)
+
+        self.conv_net = nn.Sequential(
+            nn.Conv2d(3, 16, kernel_size=3, padding=1), nn.ReLU(), nn.Conv2d(16, 32, kernel_size=3, padding=1),
+            nn.ReLU()
+        )
+        self.conv_fc_net = nn.Sequential(nn.Flatten(), nn.Linear(3200, 64), nn.ReLU())
+
+        self.fc_net_2 = nn.Sequential(nn.Linear(2, 16), nn.ReLU(), nn.Linear(16, 32), nn.ReLU(), nn.Flatten())
+
+    def forward(self, inputs: dict):
```

Comment:
add return typing, and we should try to add typing as clear as possible, such as `Dict[str, Tensor]`

Reply:
fix